config/routes.rb

```diff
@@ -1,5 +1,14 @@
 Rails.application.routes.draw do
+  get 'oauths/oauth'
+  get 'oauths/callback'
+  post "oauth/callback" => "oauths#callback"
+  get "oauth/callback" => "oauths#callback" # for use with Github, Facebook
+  get "oauth/:provider" => "oauths#oauth", :as => :auth_at_provider
+
   root "home#index"
   post "check_translation" => "home#check_translation"
-  resources :cards
+  get 'login' => 'user_sessions#new', :as => :login
+  post 'login', to: 'user_sessions#create'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.